x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/repository/CcrRepository.java

```diff
@@ -393,116 +381,64 @@ protected void restoreFiles(List<FileInfo> filesToRecover, Store store) throws I
                 final LocalCheckpointTracker requestSeqIdTracker = new LocalCheckpointTracker(NO_OPS_PERFORMED, NO_OPS_PERFORMED);
                 final AtomicReference<Tuple<StoreFileMetaData, Exception>> error = new AtomicReference<>();
 
-                final ArrayDeque<FileInfo> remainingFiles = new ArrayDeque<>(filesToRecover);
-                final Map<FileInfo, FileSession> inFlightRequests = new HashMap<>();
-                final Object mutex = new Object();
-
-                while (true) {
+                outer: for (FileInfo fileInfo : filesToRecover) {
                     if (error.get() != null) {
-                        break;
+                        break outer;
```

Comment:
I think we can remove this check here since the check inside the while loop is good enough.